core/trino-main/src/main/java/io/trino/connector/system/MaterializedViewDto.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.connector.system;
+
+import io.trino.metadata.QualifiedObjectName;
+import io.trino.spi.connector.ConnectorMaterializedViewDefinition;
+import io.trino.spi.connector.MaterializedViewFreshness;
+
+public class MaterializedViewDto
+{
+    // required because optional catalog and schema names
+    // are flaky in ConnectorMaterializedViewDefinition.
```

Comment:
What do you mean by flaky here ? We should not need this wrapper class

Reply:
ConnectorMaterializedViewDefinition`'s schema and catalog fields `Optional<String>` are empty (the object is returned from `MetadataListing`), so we need to pass a separate name along.

Reply:
ConnectorMaterializedViewDefinition's schema and catalog fields come from the session which was used to create the MV, we don't need to look at those fields. We can get the Qualified name from materializedViewPrefix.getCatalogName() and the SchemaTableName returned by getMaterializedViews